File: Python/Programmers/Level 1/x만큼 간격이 있는 n개의 숫자.py
# def solution(x, n):
#     answer = []
#     num = 1
#     while len(answer) != n:
#         answer.append(x * num) # x부터 시작해서 x씩 증가
#         num += 1
        
#     return answer

# def solution(x, n):
#     return [(i+1) * x for i in range(n)] # 길이가 n 이어야 하므로 n번 반복하면서 x씩 증가하게 함

def solution(x, n):
    # range 함수를 이용하여 숫자 범위를 만들고 리스트로 변환
    # 범위의 마지막 값을 x * (n + 1)로 하여 양수, 음수일때 처리되도록함
    # x가 0인 경우도 있으므로 0인 경우는 [0] * n을 return
    return list(range(x, x * (n + 1), x)) if x else [0] * n

File: JavaScript/Programmers/Level 1/x만큼 간격이 있는 n개의 숫자.js
// x부터 시작하여 x씩 증가하는 숫자를 n개 가진 배열을 만들면 됨

// function solution(x, n) {
//     const answer = [];
//     for (let i = 1; i <= n; i++) {
//         answer.push(i * x); // i * x를 하게 되면 x씩 증가하는 숫자가 됨 즉, (현재 인덱스 + 1) * x
//     }
    
//     return answer;
// }

// function solution(x, n) {
//     // new Array를 이용하여 크기가 n인 배열을 만든 후, 모든 요소를 0으로 채움
//     // 이후, map 메소드를 이용하여 각 요소를 (각 요소의 인덱스 + 1) * x로 mapping
//     return new Array(n).fill(0).map((_, i) => (i + 1) * x);
// }

function solution(x, n) {
    // 길이가 n인 배열을 만들고 x로 채움
    // map 메소드를 이용하여 x씩 증가하게 만듦
    return Array(n).fill(x).map((num, index) => num * (index + 1));;
}
